app/components/search_result/results_metadata_component.rb

```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+module SearchResult
+  class ResultsMetadataComponent < ViewComponent::Base
+    def initialize(document:, template_fields: [])
+      @document = document
+      @template_fields = template_fields
+      super()
+    end
+
+    def present_fields
+      @present_fields = {}
+      @template_fields.each do |field|
+        field_value = document.public_send(field)
+        if field_value.present?
+          @present_fields[field] = field_value
+        end
+      end
+      @present_fields
```

Comment:
```suggestion
      @present_fields ||= @template_fields.to_h do |field|
        [field, document.public_send(field)]
      end.compact_blank
```

Reply:
This didn't work because Array.to_h doesn't work but I was able to do what you were indicating with @template_fields.map